Percent of total in leaderboards

bcolloran opened the pull request:
add percent of total to leaderboards

bcolloran commented:
@magorlick and @jkhwu, the leaderboard part of the percent-of-total work is ready for experience review.
A few things to note:
- this is *just* the leaderboard part -- I'll begin working on the dimension table next. My recommendation is that we move ahead with merging this pr before the dimension table work is complete because this adds value and is relatively independent of that.
- this percent-of-total toggle state will not be reflected in the URL. @AdityaHegde, I will need your help with that. My recommendation is that we follow up with a second PR once this is merged; IMO it's better to have this functionality sooner without the URL state than to delay merging in order to finish that detail
- remember that, per the spec, you will need to add `valid_percent_of_total: true` to any measures for which you want to see the percent of total

AdityaHegde commented on web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte:
```diff
@@ -81,6 +106,53 @@
     showHideDimensions.setAllToVisible();
   };
+
+  let disabledButtons: ("delta" | "pie")[] = [];
```
Should we move this to a separate component? Showing percents doesnt fit within LeaderboardMeasureSelector.

AdityaHegde commented on web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte:
```diff
@@ -19,6 +27,23 @@
+  );
+  $: hasTimeSeries = $hasTimeSeriesQuery?.data;
+
+  $: {
```
nit: remove the log

AdityaHegde commented on web-common/src/features/dashboards/leaderboard/DimensionLeaderboardEntrySet.svelte:
```diff
@@ -59,31 +67,69 @@ see more button
+    // FIXME: `showComparisonForThisValue` should not be the responsibility
+    // of this component; the handling of the on:mouseenter/on:mouseleave
+    // events should be done in each individual DimensionLeaderboardEntry
+    const showComparisonForThisValue = comparisonLabelToReveal === v.label;
```
This whole block is rerun for every `comparisonLabelToReveal` change, i.e. hovering over leaderboard values. Perhaps we should have both versions of formatted string and toggle while displaying. I see there is already a FIXME. Any idea on when we are planning to address it?

bcolloran replied:
I don't know, I was going to mention to @nishantmonu51 that these components have accumulated a great deal of tech debt and badly need some refactoring (I left several other FIXMEs in this PR in addition to this one). It is my recommendation to start on that right away, but it's up to Nishant to make the call.

bcolloran replied:
@AdityaHegde , discused with Nishant this morning-- the current plan is to tackle these FIXMES after working on the % of total for the dimension table

bcolloran commented:
Thank you for the helpful review @AdityaHegde! I moved the button toggle to a separate component as you recommended, and renamed `LeaderboardMeasureSelector` to `LeaderboardControls` to clarify that it now does more than just measure selection. This should now be ready for final code review and merge.

@nishantmonu51 and @jkhwu -- if you wish to do a QA/experience pass, this branch is ready (but see notes above)

AdityaHegde reviewed (approved):
Code wise this looks good.

nishantmonu51 commented:
Looks good to be merged to unblock other work.
One boundary case to be fixed here - 

1. Enable `valid_percent_of_total` for metric A
2. Select order by metric A and switch to showing percentage of totals 
3. Switch to a different metric B for which % of total is not enabled
4. Metric B is shown with % of totals 
Expected behavior -> Hide the % of totals column when switching to a metric for which it should not be enabled.
